### ISSUE TITLE
Ensure disposable invocation pipelines are hooked up.

### ISSUE THREAD

**Issue by macrogreg** [temporalio/sdk-dotnet#16]

**Comment by cretz** [comment]
I would suggest, if at all possible, let the caller create things they want to send us and let them control their lifetimes and when they dispose.

**Comment by macrogreg** [comment]
Good point. We solve this by letting users specify factories for the items in the pipeline, rather than the items themselves. The factories are invokes for each pipeline. Now, users can easily pick whether they return the same or a different instance for different pipelines.
